libnmstate/netapplier.py

```diff
@@ -63,13 +64,32 @@ def _apply_ifaces_state(interfaces_desired_state, verify_change):
             ifaces_current_state = _index_by_name(netinfo.interfaces())
             _edit_interfaces(ifaces_desired_state, ifaces_current_state)
         if verify_change:
-            # FIXME: Remove this sleep after adding wait for state mechanism.
-            time.sleep(2)
-            ifaces_current_state = _index_by_name(netinfo.interfaces())
-            ifaces_desired_state = _remove_absent_iface_entries(
-                ifaces_desired_state)
-            ifaces_desired_state = remove_ifaces_metadata(ifaces_desired_state)
-            assert_ifaces_state(ifaces_desired_state, ifaces_current_state)
+            verified_successfully = False
+            for retry in range(4):
+                try:
+                    _verify_change(ifaces_desired_state)
+                except DesiredStateIsNotCurrentError:
+                    time.sleep(1)
+                else:
+                    verified_successfully = True
+                    break
+            if not verified_successfully:
```

Comment:
I think this variable is not require and it should work with an `else` for the for loop. But the current way might be easier to read if one is not familiar with the "for... else..." syntax.

Reply:
Another option to implenent do..until would be to use:

while True... break. Then there would be only one call of `_verify_change` instead of two. This would make it less likely that only one of them is adjusted in the future.

Reply:
Lets use `else`, it is the python way. Thanks.